snapcraft/internal/repo.py

```diff
@@ -576,3 +568,18 @@ def _set_pkg_version(pkg, version):
         pkg.candidate = version
     else:
         raise PackageNotFoundError('{}={}'.format(pkg.name, version))
+
+
+_libc_lib_list = None
+
+
+def libc_library_list():
```

Comment:
hmm, global dicts; make sure it is non racy and works with copies at all times (more so for whatever it returns.